src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIReferenceAnalyzer.cs

```diff
@@ -71,6 +74,10 @@ public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state,
                     state.Packages.Remove(package, new OperationDetails() { Risk = BuildBreakRisk.Medium, Details = ImmutableList.Create<string>(package.Name) });
                 }
             }
+
+            var result = new Analysis.AnalyzeResult { FileLocation = "fileLocation", FullDescription = "Description", ResultMessage = "result" };
+            var resultDefinition = new Analysis.AnalyzeResultDefinition { Name = "name", AnalysisResults = ImmutableList.Create(result).ToAsyncEnumerable() };
+            await this._upgradeResultWriter.WriteAsync(ImmutableList.Create(resultDefinition).ToAsyncEnumerable(), token);
```

Comment:
this is just an example of how the consumers - steps and substeps can use this. Will be removed from this PR